lib/java_buildpack/framework/contrast_security_agent.rb

```diff
@@ -109,6 +109,10 @@ def application_name
         @application.details['application_name'] || 'ROOT'
       end
 
+      def app_name_defined?
+        @droplet.java_opts.as_env_var.include?('contrast.override.appname')
```

Comment:
The more Ruby idiomatic way to have done this was with `java_opts.any? { |java_opt| java_opt =~ 'contrast.override.appname')`.  I'll polish up in a following commit.